witchcraft-metrics/src/registry.rs

```diff
@@ -209,7 +228,9 @@ impl MetricRegistry {
     where
         T: Into<MetricId>,
     {
-        self.histogram_with(id, Histogram::default)
+        self.histogram_with(id, || {
+            Histogram::new(ExponentiallyDecayingReservoir::new_with(self.clock.clone()))
```

Comment:
It's unfortunate we have to copy the default logic for Histogram here.

Reply:
Yeah it just didn't seem like a great idea to add yet another constructor to Histogram.